cql3/statements/create_index_statement.cc

```diff
@@ -202,6 +206,34 @@ void create_index_statement::validate_for_frozen_collection(const index_target&
     }
 }
 
+void create_index_statement::validate_for_collection(const index_target& target, const column_definition& cd) const
+{
+    auto throw_exception = [&] {
+        const char* msg_format = "Cannot create secondary index on {} of non-frozen collection column {}";
+        throw exceptions::invalid_request_exception(format(msg_format, to_sstring(target.type), cd.name_as_text()));
+    };
+    switch (target.type) {
+        case index_target::target_type::regular_values:
+        case index_target::target_type::full:
+            throw_exception();
+        case index_target::target_type::keys:
+            if (!(cd.type->is_map() || cd.type->is_set())) {
```

Comment:
isn't it more intuitive to write !cd.type->is_map() && !cd.type->is_set() ? compiler should be smart nowadays to optimize this, assuming the way you wrote it saves a few instructions.

Reply:
It is, it's just de Morgan's laws which flow my veins.